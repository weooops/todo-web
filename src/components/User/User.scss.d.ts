export const user: string;
export const red: string;
