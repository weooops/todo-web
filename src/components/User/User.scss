.user {
  font-weight: bold;
}

.red {
  color: red;
}