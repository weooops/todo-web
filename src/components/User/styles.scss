.user {
  font-weight: bold;
  color: purple;
}
