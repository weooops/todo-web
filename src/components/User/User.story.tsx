import * as React from 'react';
import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import User from './User';

storiesOf('User', module)
  .addDecorator(withKnobs)
  .add('default', () => {
    const props = {
      user: {
        id: 1,
        username: text('username', 'jake')
      }
    };
    return <User {...props} />;
  });