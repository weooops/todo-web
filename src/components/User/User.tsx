import * as React from 'react';

import * as styles from './styles.scss';
import * as classNames from 'classnames/bind';
import { UserType } from '../../models';

const cx = classNames.bind(styles);

interface PropsType {
  user: UserType;
}

class User extends React.Component<PropsType> {
  public shouldComponentUpdate(nextProps: PropsType): boolean {
    return nextProps.user !== this.props.user;
  }

  public render(): JSX.Element {
    const { user: { username } } = this.props;
    
    return (
      <div className={cx(styles.user)}>
        {username}
      </div>
    );
  }
}

export default User;