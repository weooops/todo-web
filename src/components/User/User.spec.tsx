import * as React from 'react';
import { shallow } from 'enzyme';

import User from './User';

it('renders correctly', () => {
  const props = {
    user: {
      id: 1,
      username: 'jake'
    }
  };
  const wrapper = shallow(<User {...props} />);
  expect(wrapper).toMatchSnapshot();
});