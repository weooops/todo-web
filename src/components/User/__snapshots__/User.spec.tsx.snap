// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <User
    user={
      Object {
        "id": 1,
        "username": "jake",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "jake",
      "className": "",
    },
    "ref": null,
    "rendered": "jake",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "jake",
        "className": "",
      },
      "ref": null,
      "rendered": "jake",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
